### ISSUE TITLE
MET systematics

### ISSUE THREAD

**Issue by jbossios** [UCATLAS/xAODAnaHelpers#1364]
Hello,

I'm trying to get MET systematics in my TTree but I am having no success.

c.algorithm("METConstructor", {
  "m_name"                        : "met",
  "m_referenceMETContainer"  : "MET_Reference_AntiKt4EMTopo",
  "m_mapName"                     : "METAssoc_AntiKt4EMTopo",
  "m_coreName"                    : "MET_Core_",
  "m_outputContainer"           : "MET",
  "m_inputMuons"                  : "Muons_Selected",
  "m_inputJets"                      : "Jets_Selected",
  "m_runNominal"                  : False,
  "m_doJVTCut"                    : True,
  "m_dofJVTCut"                   : False,
  "m_calculateSignificance"       : False,
  "m_jetSystematics"                  : "All",
  "m_muonSystematics"             : "All",
  "m_outputAlgoSystNames"      : "MET_Algo",
  } )

I have the following in TreeAlgo:

  "m_METContainerName"            : "MET",
  "m_metSystsVec"                 : "MET_Algo",

Any idea what I am missing? I see several TTrees for the muon and jet systematics but not for MET.

Best,
Jona

**Comment by kratsg** [comment]
What does the log output look like when running the algorithm? Should get something like https://github.com/UCATLAS/xAODAnaHelpers/blob/c95e847f56a7b42cea647f2cefade19fea623a2b/Root/METConstructor.cxx#L160-L163

You should also be seeing (if you enable debug) loops over systematics in the constructor: https://github.com/UCATLAS/xAODAnaHelpers/blob/c95e847f56a7b42cea647f2cefade19fea623a2b/Root/METConstructor.cxx#L279

Then if we know for sure the met constructor algorithm is doing the right thing, then we can go ahead and check if the TreeAlgo is picking it up correctly.

**Comment by jbossios** [comment]
Hi @kratsg 

Thanks for the hep. I have MET systematics when using EMTopo jets, although I'm still trying to figure out how to get the systematics for PFlow jets. I created https://github.com/UCATLAS/xAODAnaHelpers/pulls to be able to use p-flow jets in METMaker. I will create another PR once I learn how to change the config in METSystematicTool (I tried a couple of things but didn't work, I sent an email to a couple of JetEtmiss lists).
